### DISCUSSION TITLE
Registry Values Not Updating Correctly with WIX 4.0.4 HotFix MSI and WIX 3.14 Base MSI"

### DISCUSSION THREAD

**Discussion by kalyani1217** [wixtoolset/issues#8123] [Questions]
Hi,
I've developed a Base MSI using WIX 3.14 and a HotFix MSI using WIX 4.0.4. Subsequently, I created a patch by combining the Base MSI and HotFix MSI, resulting in an MSP file.

However, upon applying the MSP file over the Base MSI, I noticed that the registry values were not updated; they still reflect the values from the Base MSI instead of those intended from the HotFix MSI.

![image](https://github.com/wixtoolset/issues/assets/29778816/5e512fda-3221-4515-885b-1834645e4c81)

In the provided image, the BATestVersion registry entry should have been updated to 07.00.00.0101 by the HotFix MSI, but it remains 07.00.00.0000. Additionally, I removed the InputString and ResultString variables, yet they persist. However, any newly added values appear correctly.

Could you please advise on why the values from the Base MSI are being retained instead of reflecting those from the HotFix MSI?

**Comment by robmen** [comment]
Using different major versions of WiX to build and patch your MSI is not likely to work. Patching is hypersensitive to any changes in the deep internals of how the MSI is structured. Advancements in major WiX versions have a high probability of disrupting patching.

Didn't work on WiX v2 to v3. Not likely to work v3 to v4.
